aiida/backends/general/abstractqueries.py

```diff
@@ -185,56 +217,22 @@ def get_bands_and_parents_structure(self, args):
             if already_visited_bdata.__contains__(bid):
                 continue
             already_visited_bdata.add(bid)
-
-            if args.element is not None:
-                all_symbols = [_['symbols'][0] for _ in akinds]
-                if not any([s in args.element for s in all_symbols]
-                           ):
-                    continue
-
-            if args.element_only is not None:
-                all_symbols = [_['symbols'][0] for _ in akinds]
-                if not all(
-                        [s in all_symbols for s in args.element_only]
-                ):
-                    continue
-
-            # We want only the StructureData that have attributes
-            if akinds is None or asites is None:
+            formula = self._extract_formula(args, akinds, asites)
+            if formula is None:
                 continue
-
-            symbol_dict = {}
-            for k in akinds:
-                symbols = k['symbols']
-                weights = k['weights']
-                symbol_dict[k['name']] = get_symbols_string(symbols,
-                                                            weights)
-
-            try:
-                symbol_list = []
-                for s in asites:
-                    symbol_list.append(symbol_dict[s['kind_name']])
-                formula = get_formula(symbol_list,
-                                      mode=args.formulamode)
-            # If for some reason there is no kind with the name
-            # referenced by the site
-            except KeyError:
-                formula = '<<UNKNOWN>>'
-            entry_list.append([str(bid), str(formula),
-                               bdate.strftime('%d %b %Y'), blabel])
+            entry_list.append([str(bid), str(formula), bdate.strftime('%d %b %Y'), blabel])
 
         return entry_list
 
-    def get_all_parents(self, node_pks, return_values=('id',)):
-        """
-        Get all the parents of given nodes
+    @staticmethod
+    def get_all_parents(node_pks, return_values=('id',)):
+        """Get all the parents of given nodes
+
         :param node_pks: one node pk or an iterable of node pks
-        :return: a list of aiida objects with all the parents of the nodes
-        """
+        :return: a list of aiida objects with all the parents of the nodes"""
         from aiida.orm import Node, QueryBuilder
 
-        qb = QueryBuilder()
-        qb.append(Node, tag='low_node',
-                  filters={'id': {'in': node_pks}})
-        qb.append(Node, with_descendants='low_node', project=return_values)
-        return qb.all()
+        q_bld = QueryBuilder()
```

Comment:
Not really important, but perhaps it would be good to use a standard name for QB instances since they are used in so many places.
q_bld, q_build or qbld would all be fine... anyhow perhaps just replace q_build by q_bld (or vice versa) for the time being

Reply:
done